### PR TITLE
BS-14694 Add dependency on web extension project

### PR DESCRIPTION
**Pull request by rbioteau**
* Remove interfaces from console-server
* Only keep implementations
* Support new RestAPIContoller and PageController in renderers

### CODE REVIEW

File: server/src/main/java/org/bonitasoft/console/common/server/page/CustomPageService.java [hunk 1]
```diff
@@ -120,13 +121,13 @@ public void ensurePageFolderIsUpToDate(final APISession apiSession, final PageRe
     }
 
     @SuppressWarnings("unchecked")
-    public Class<PageController> registerPage(final GroovyClassLoader pageClassLoader, final PageResourceProvider pageResourceProvider)
+    public Class<PageController> registerPage(final GroovyClassLoader pageClassLoader, final PageResourceProviderImpl pageResourceProvider)
```
Comment (apupier) [inline server/src/main/java/org/bonitasoft/console/common/server/page/CustomPageService.java]: Why are you asking for an Impl now? We shoudl still use the interface, no?
BTW, if the class ha sbeen updated, and the new interface doesn't inherits from the old one, we need to provide both methods and deprecates the old one

Comment (rbioteau) [inline server/src/main/java/org/bonitasoft/console/common/server/page/CustomPageService.java]: So I have update signature to use interface now.
However, there was some protected method in the old implementation that was use in the service that we do not want to expose in the api. That's why there is still some Impl visible in the code.

**Comment by rbioteau** [comment]
Up

**Comment by rbioteau** [comment]
Up

**Pull request merged**